platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ImportMigrationConfigurationAdminContext.java

```diff
@@ -0,0 +1,312 @@
+        // categorize exported configurations
+        final ImportMigrationConfigurationAdminEntry[] entries = context.entries()
+                .map(this::proxy)
+                .filter(Objects::nonNull)
+                .toArray(ImportMigrationConfigurationAdminEntry[]::new);
+
+        this.exportedServices = Stream.of(entries)
+                .filter(ImportMigrationConfigurationAdminEntry::isManagedService)
+                .collect(Collectors.toMap(ImportMigrationConfigurationAdminEntry::getPid,
+                        Function.identity()));
+        this.exportedFactoryServices = Stream.of(entries)
+                .filter(ImportMigrationConfigurationAdminEntry::isManagedServiceFactory)
+                .collect(Collectors.groupingBy(ImportMigrationConfigurationAdminEntry::getFactoryPid));
+        context.getReport()
+                .doAfterCompletion(this::deleteUnexportedConfigurationsAfterCompletion);
+    }
+
+    private static boolean isManagedServiceFactory(Configuration cfg) {
+        return cfg.getFactoryPid() != null;
+    }
+
+    private static boolean isManagedService(Configuration cfg) {
+        return cfg.getFactoryPid() == null;
+    }
+
+    @Override
+    public Stream<ImportMigrationEntry> entries() {
+        if (!isValid) {
+            return Stream.empty();
+        }
+        return Stream.concat(exportedServices.values()
+                        .stream(),
+                exportedFactoryServices.values()
+                        .stream()
+                        .flatMap(List::stream));
+    }
+
+    @Override
+    public ImportMigrationEntry getEntry(Path path) {
+        throw new IllegalStateException("should not be called");
+    }
+
+    @Override
+    public Stream<ImportMigrationEntry> entries(Path path) {
+        throw new IllegalStateException("should not be called");
+    }
+
+    @Override
+    public Stream<ImportMigrationEntry> entries(Path path, PathMatcher filter) {
+        throw new IllegalStateException("should not be called");
+    }
+
+    // PMD.DefaultPackage - designed to be called only from ImportMigrationConfigurationAdminEntry within this package
+    @SuppressWarnings("PMD.DefaultPackage")
+    Configuration createConfiguration(ImportMigrationConfigurationAdminEntry entry)
+            throws IOException {
+        // Question: should we use the bundle location that was exported???
+        if (entry.isManagedServiceFactory()) {
+            return configurationAdmin.createFactoryConfiguration(entry.getFactoryPid(), null);
+        }
+        return configurationAdmin.getConfiguration(entry.getPid());
+    }
+
+    // PMD.DefaultPackage - designed to be called only from ImportMigrationConfigurationAdminEntry within this package
+    @SuppressWarnings("PMD.DefaultPackage")
+    Configuration getMemoryConfiguration(ImportMigrationConfigurationAdminEntry entry) {
+        final String fpid = entry.getFactoryPid();
+
+        if (fpid != null) {
```

Comment:
✏️ Inverting this test will reduce the complexity of the method.

Reply:
???